components/esp_dns/README.md

```diff
@@ -0,0 +1,189 @@
+| Supported Targets | ESP32 | ESP32-C2 | ESP32-C3 | ESP32-C5 | ESP32-C6 | ESP32-C61 | ESP32-H2 | ESP32-P4 | ESP32-S2 | ESP32-S3 |
+| ----------------- | ----- | -------- | -------- | -------- | -------- | --------- | -------- | -------- | -------- | -------- |
+
+# ESP DNS Component
+
+This component provides a flexible DNS resolution system for ESP32 devices with support for multiple DNS protocols. It allows applications to securely resolve domain names using various transport methods, including standard UDP/TCP DNS, DNS over TLS (DoT), and DNS over HTTPS (DoH).
+
+## Table of Contents
+- [Features](#features)
+- [How It Works](#how-it-works)
+- [Certificate Options](#certificate-options)
+- [How to Use](#how-to-use)
+- [Configuration](#configuration)
+- [Common DNS Servers](#common-dns-servers)
+- [Requirements](#requirements)
+- [Limitations](#limitations)
+- [Performance Considerations](#performance-considerations)
+- [Troubleshooting](#troubleshooting)
```

Comment:
```suggestion
- [Features](#features)
- [Requirements](#requirements)
- [How to Use](#how-to-use)
- [Configuration](#configuration)
- [Certificate Options](#certificate-options)
- [Common DNS Servers](#common-dns-servers)
- [Limitations](#limitations)
- [Performance Considerations](#performance-considerations)
- [How It Works](#how-it-works)
- [Troubleshooting](#troubleshooting)
```

Reply:
Reorder the outline for a better reading flow.

Reply:
Done